app/src/main/java/com/marvelapp/presentation/activity/MarvelActivity.kt

```diff
@@ -0,0 +1,39 @@
+package com.marvelapp.presentation.activity
+
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.marvelapp.data.MarvelStates
+import com.marvelapp.databinding.ActivityMainBinding
+import com.marvelapp.presentation.view.MarvelView
+import com.marvelapp.presentation.viewModel.MarvelViewModel
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class MarvelActivity : AppCompatActivity() {
```

Comment:
You should separate the SplashActivity and the HomeActivity in both differents screens